File: okpermission/src/main/java/com/wings/okpermission/javav2/PermissionX.java
package com.wings.okpermission.javav2;

import android.content.Context;
import android.content.pm.PackageManager;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.FragmentActivity;

/**
 * @author AI
 */
public class PermissionX {

    public static PermissionCollection init(FragmentActivity activity) {
        return new PermissionCollection(activity);
    }

    public static boolean isGranted(Context context, String permission) {
        return ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED;
    }

}

File: okpermission/src/main/java/com/wings/okpermission/kotlinv2/PermissionX.kt
package com.wings.okpermission.kotlinv2

import android.content.Context
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat
import androidx.fragment.app.FragmentActivity

/**
 * An open source Android library that makes handling runtime permissions extremely easy.
 *
 * The following snippet shows the simple usage:
 * ```kotlin
 *   PermissionX.init(activity)
 *      .permissions(Manifest.permission.READ_CONTACTS, Manifest.permission.CAMERA)
 *      .request { allGranted, grantedList, deniedList ->
 *          // handling the logic
 *      }
 *```
 *
 * @author AI
 */
object PermissionX {

    fun init(activity: FragmentActivity) : PermissionCollection {
        return PermissionCollection(activity)
    }

    fun isGranted(context: Context, permission: String) : Boolean {
        return ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED
    }

}

